traffic_ops/traffic_ops_golang/topology/topologies.go

```diff
@@ -150,6 +150,19 @@ func (topology *TOTopology) Validate() error {
 	rules["empty cachegroups"] = topology_validation.CheckForEmptyCacheGroups(topology.ReqInfo.Tx, cacheGroupIds, dsCDNs, false, nil)
 	rules["required capabilities"] = topology.validateDSRequiredCapabilities()
 
+	for _, cdnId := range dsCDNs {
+		err, dsId := dbhelpers.GetDSByCDNIdTopology(topology.ReqInfo.Tx.Tx, cdnId, topology.Name)
```

Comment:
Instead of looping over each CDN and running the check for each CDN, we should just pass the `dsCDNs` slice into `CheckTopologyOrgServerCGInDSCG` and update its query to replace `WHERE ds.deliveryservice =ANY($1)` with `WHERE ds.cdn_id = ANY($1)`. We would also then have to add `INNER JOIN topology top ON ds.topology = top.name` and add `AND top.name = $3` to the `WHERE` clause. That would allow us to get all the data in a single query, and we could delete the `GetDSByCDNIdTopology` function.

Reply:
This `INNER JOIN topology top ON ds.topology = top.name` doesn't make any sense, because `topology` table doesn't have any column that would help with delivery service results

Reply:
Ah, it's hard to come up with good SQL queries in a review comment. You're right. The `INNER JOIN topology` would not be required, since you can add `AND d.topology = $3` to the `WHERE` clause instead.